player/lua/select.lua

```diff
@@ -261,54 +261,92 @@ mp.add_key_binding(nil, "select-edition", function ()
+                sub_content[time_seconds] = {}  -- Initialize content list for that timestamp
+            end
 
-    if sub == nil then
-        show_warning("No subtitle is loaded.")
-        return
-    end
+            -- If the subtitle content is not already in the list for that timestamp, add it
+            local content_exists = false
+            for _, exist_content in ipairs(sub_content[time_seconds]) do
+                if exist_content == sub_line then
+                    content_exists = true
+                    break
+                end
+            end
 
-    if sub.external and sub["external-filename"]:find("^edl://") then
-        sub["external-filename"] = sub["external-filename"]:match('https?://.*')
-                                   or sub["external-filename"]
+            if not content_exists then
+                sub_content[time_seconds][#sub_content[time_seconds] + 1] = sub_line
+            end
+        end
     end
 
-    local r = mp.command_native({
-        name = "subprocess",
-        capture_stdout = true,
-        args = sub.external
-            and {"ffmpeg", "-loglevel", "error", "-i", sub["external-filename"],
-                 "-f", "lrc", "-map_metadata", "-1", "-fflags", "+bitexact", "-"}
-            or {"ffmpeg", "-loglevel", "error", "-i", mp.get_property("path"),
-                "-map", "s:" .. sub["id"] - 1, "-f", "lrc", "-map_metadata",
-                "-1", "-fflags", "+bitexact", "-"}
-    })
+    -- Process all timestamps and content into selectable subtitle list
+    for time_seconds, contents in pairs(sub_content) do
+        for _, sub_line in ipairs(contents) do
+            sub_times[#sub_times + 1] = time_seconds
+            sub_lines[#sub_lines + 1] = format_time(time_seconds, duration) .. " " .. sub_line
+        end
+    end
 
-    if r.error_string == "init" then
-        show_error("Failed to extract subtitles: ffmpeg not found.")
-        return
-    elseif r.status ~= 0 then
-        show_error("Failed to extract subtitles.")
-        return
+    -- Sort by timestamp
+    local function compare_times(a, b)
+        return a < b
+    end
+    local sorted_sub_times = {}
+    for i = 1, #sub_times do
+        sorted_sub_times[i] = sub_times[i]
     end
+    table.sort(sorted_sub_times, compare_times)
 
-    local sub_lines = {}
-    local sub_times = {}
-    local default_item
-    local delay = mp.get_property_native("sub-delay")
-    local time_pos = mp.get_property_native("time-pos") - delay
-    local duration = mp.get_property_native("duration", math.huge)
+    -- Use a helper table to avoid duplicates
+    local added_times = {}
+
+    -- Rebuild sub_lines and sub_times based on the sorted timestamps
+    local sorted_sub_lines = {}
+    for _, sub_time in ipairs(sorted_sub_times) do
+        -- Iterate over all subtitle content for this timestamp
+        -- keeping only unique subtitles for each timestamp
+        if not added_times[sub_time] then
```

Comment:
This seems to break multi-line subtitle lines.

Reply:
Oh nevermind you're still keeping different lines with the same timestamp.

Reply:
The comment here is wrong, it actually iterates through all the different subtitle contents that hold the same time stamp
![image](https://github.com/user-attachments/assets/df2d1ba1-c711-4a7d-9c6d-3eb763a27f96)